File: SH/etc/11. Container_With_Most_Water.py
def maxArea(height: List[int]) -> int:
    area = 0
    start, end = 0, len(height)-1
    while(start<end):
        tmp_b = end-start
        tmp_h = min(height[start], height[end])
        tmp_a = tmp_b * tmp_h
        area = max(area, tmp_a)
        if height[start] < height[end]:
            start += 1
        else:
            end -= 1
    return area

File: SH/etc/11. Container_With_Most_Water.c



// Old Solution
int maxArea(int* height, int heightSize){
    int l = 0, r = heightSize - 1;
    int maxArea = 0;
    while ( l < r ){
        int area = height[l] < height[r] ? height[l] * (r-l) : height[r] * (r-l);
        if ( area > maxArea ){
            maxArea = area;
        } 
        
        if ( height[l] < height[r]) l++;
        else r--;
    }
    return maxArea;
}